matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java

```diff
@@ -0,0 +1,61 @@
+package org.matsim.core.router;
+
+import org.apache.log4j.Logger;
+import org.matsim.api.core.v01.Id;
+import org.matsim.api.core.v01.network.Link;
+import org.matsim.api.core.v01.network.Network;
+import org.matsim.core.gbl.Gbl;
+import org.matsim.core.network.NetworkUtils;
+import org.matsim.facilities.FacilitiesUtils;
+import org.matsim.facilities.Facility;
+
+class MultimodalLinkChooserDefaultImpl implements MultimodalLinkChooser {
```

Comment:
So, this is some complicated logic. I don't see any unit test for this. Are you testing this implicitly somewhere else?

Reply:
Basically it is tested in the Leipzig [scenario](https://github.com/matsim-scenarios/matsim-leipzig/commit/4cb6fbcdb0fca84185747b1f1220bb68f224b435).

Reply:
I think we need some test in matsim-libs, too

Reply:
> Basically it is tested in the Leipzig [scenario](https://github.com/matsim-scenarios/matsim-leipzig/commit/4cb6fbcdb0fca84185747b1f1220bb68f224b435).

That's not sufficient I think.

Reply:
I will write a test for this class, which covers all the possible outcomes of the logic.

Reply:
Thanks for doing this. Sorry that you have to do it now. This should have been there in the old place already.